### ISSUE TITLE
Serialization: Field filter modification on source Instance doesn't correctly update the Field Filter on Destination Instance

### ISSUE THREAD

**Issue by ixipixi** [metabase/metabase#41255]
### Describe the bug

If you have a field filter set up on a source instance and destination instance that your managing with serialization for version control, changing the column reference on the source instance, exporting the change and importing into the destination doesn't result in correct values being displayed in the field filter.

### To Reproduce

1. Set up 2 fresh Metabase installation
2. In Instance1 create "Test Dashboard" with a question that has a field filter on it that populates a list of values from another question
3. Create an export
4. Import this into Instance2
5. Note that the imported "Test Dashboard" and field filter works in Instance2 as expected
6. Go back to Instance1 and edit the field filter on "Test Dashboard" to refer to a different text column
7. Create a new export
8. Import this into Instance2 to update the dashboard
9. Go to "Test Dashboard" and note that it doesn't show the values from the newly selected text column - it still displays the original list from prior to the reimport. You also cannot edit the field filter - Metabase doesn't recognize t4ext columns exist on the table the filter is source from. navigating to the question the values are sourced from corrects this issue.|

Loom: https://www.loom.com/share/fcfa604a9f9a495881235563fc68506e

### Expected behavior

Changes from the source instance should reflect correctly in the destination instance.

### Logs

HAR file navigating the dashboard and opening the edit modal in Instance2 after the field filter has been broken.

[har.zip](https://github.com/metabase/metabase/files/14932923/har.zip)

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.22+7",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.22",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.22+7",
    "os.name": "Linux",
    "os.version": "5.15.133.1-microsoft-standard-WSL2",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.22 (Debian 11.22-1.pgdg110+1)"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.2"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-03-14",
      "tag": "v1.49.0",
      "hash": "46c668b"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

For customers that use serialization for version control this is significant.

**Comment by ixipixi** [comment]
Also confirmed this is still an issue in 49.3

**Comment by piranha** [comment]
@ixipixi I wasn't able to reproduce that, could you please verify that I'm not missing anything peculiar: https://www.loom.com/share/3b0aee6aee4345c5b811d6be0365b1fd

This is on master but I believe it has no meaningful difference regarding serialization to any of latest releases.

**Comment by ixipixi** [comment]
@piranha in my test case the dashboard filter was set up to pull a list of values from another model or question. Changing which column the values come from in the source and then importing that change is where the bug is introduced. 

![filter_change](https://github.com/metabase/metabase/assets/13661163/c42b20a7-d006-40ef-be51-cec535e3dad0)

**Comment by piranha** [comment]
@ixipixi okay great news! I can reproduce exactly this issue on 49.0, but not on master - which means we have it fixed, I just need to find that fix. :)

**Comment by piranha** [comment]
@ixipixi this is a problem with [result_metadata not being serialized](https://github.com/metabase/metabase/pull/41131) and is fixed in 0.49.5.

**Comment by ixipixi** [comment]
@piranha awesome. Thank you!
